doubly_linked_list/doubly_linked_list.py

```diff
@@ -44,25 +44,95 @@ def __len__(self):
     return self.length
 
   def add_to_head(self, value):
-    pass
+    new_node = ListNode(value)
+    self.length += 1
+    if self.tail is None and self.head is None:
+      self.tail = new_node
+      self.head = new_node
+    else:
+      new_node.next = self.head
+      self.head.prev = new_node
+      self.head = new_node
 
   def remove_from_head(self):
-    pass
+    if self.head is None and self.tail is None:
+      return None
+    
+    self.length -= 1
+    if self.head == self.tail :
+      current_head = self.head
+      self.head = None
+      self.tail = None
+      return current_head.value
+
+    current_head = self.head
+    self.head = self.head.next
+    self.head.prev = None
+    return current_head.value
 
   def add_to_tail(self, value):
-    pass
+    new_node = ListNode(value)
+    self.length += 1
+    if self.tail is None and self.head is None:
+      self.tail = new_node
+      self.head = new_node
+    else:
+      new_node.prev = self.tail
+      self.tail.next = new_node
+      self.tail = new_node
 
   def remove_from_tail(self):
-    pass
+    if not self.head and not self.tail :
+      return None
+    
+    self.length -= 1
+    if self.head == self.tail :
+      current_tail = self.tail
+      self.head = None
+      self.tail = None
+      return current_tail.value
+
+    current_tail = self.tail
+    self.tail = self.tail.prev
+    self.tail.next = None
+    return current_tail.value
+    
 
   def move_to_front(self, node):
-    pass
+    if node == self.tail:
+      self.tail = self.tail.prev
+
+    self.head.insert_before(node.value)
+    self.head = self.head.prev
+    node.delete()
 
   def move_to_end(self, node):
-    pass
+    if node == self.head:
+      self.head = self.head.next
+
+    self.tail.insert_after(node.value)
+    self.tail = self.tail.next
+    node.delete()
 
   def delete(self, node):
-    pass
+    if node == self.head :
+      self.head = self.head.next
+      
+    if node == self.tail :
+      self.tail = self.tail.prev
+      
+    node.delete()
+    self.length -= 1
     
   def get_max(self):
-    pass
+    current_node = self.head
+    max_value = 1
+    while current_node != self.tail:
+      if current_node.value > current_node.next.value :
+        max_value = current_node.value
+      else:
+        max_value = current_node.next.value
+      
+      current_node = current_node.next
+    return(max_value)
+
```

Comment:
could cut down a few lines of code by comparing next.value to our current max value and only change the max value if our next.value is greater